### DISCUSSION TITLE
Add a tabs in the Markdown in profile the of GitHub

### DISCUSSION THREAD

**Discussion by ghost** [community/community#14964] [Other Feature Feedback, Questions, & Ideas]
Hi Community!

#### 1. Feature name
Add a tabs in the Markdown in profile the of Github

#### 2. Feature description 
1. I think it would be nice to have markdown tab support in your Github user profile if you have a lot of information to show. 
2. According to this official link:  [GitHub Flavored Markdown Spec](https://github.github.com/gfm/)  - here it does not say that there is support for tabs in markdown. But there is an implementation called rmarkdown that supports tabs in Markdown. 
3. Would find it interesting if there was the possibility of supporting tabs in the Github user profile as I show in this photo here:

#### 3. ui/ux
![image](https://user-images.githubusercontent.com/95204402/167318895-424e5fcc-884f-44dd-bd81-4ba096e02316.png)

#### 4. Source-code "github.com/username/username/README.md"
```markdown 
## title {.tabset .tabset-fade}
content above tabbed region.

### tab Social-Media

tab content Social-Media

### tab Contact

tab content  Contact

### tab Revisions

tab content  Revisions

### tab Articles

tab content  Articles

### tab Skills

tab content  Skills

### tab Insights

tab content  Insights

content below tabbed region
```

#### 5. user stories
I would like as a user to create a README.md for my user profile with the information per tab

#### 6. Notes
1. I like erudite music. My profile picture is my tribute to Ludwig van Beethoven and "draco dormiens nunquam titillandus" - reference to Harry Potter(Hogwarts).
2. Hope to help Github Community.
3. The references I left about the software are not to promote any software company or product, they serve as a bibliographic reference. So, I just want to show you things I've read, researched, tested in this references. 
4. And I didn't find anything talking about this feature... I didn't find any requests for this feature.

#### 7. This would mean:
- GH markdown is more compatible with other solutions.
- You aren't introducing yet another non-standard markdown feature
- It is more future proof. 

#### 8. References
1. [pymdown-extensions](https://facelessuser.github.io/pymdown-extensions/extensions/tabbed/)
2. [tabbed-sections](https://bookdown.org/yihui/rmarkdown/html-document.html#tabbed-sections)
5. [rpubs](https://rpubs.com/sunzhe96/596831)
6. [stackoverflow - rmarkdown tabbed content](https://stackoverflow.com/questions/38062706/rmarkdown-how-to-end-tabbed-content)
7. [docusaurus](https://docusaurus.io/docs/markdown-features/admonitions )

**Comment by ghost** [comment]
great idea!

**Comment by Thiamath** [comment]
This will work great when documenting how to run an application in different systems!

**Reply by ghost** [reply]
Hi Thiamath. What do you think about these use cases?

#### example and use case 1: Add tabbed code snippets in Markdown
![image](https://user-images.githubusercontent.com/111701513/210266071-3dc486ce-2cd8-4695-8deb-a5e66b0f7006.png)

#### example and use case 2: Add social media links or add tabs in the Markdown in profile the of GitHub
![image](https://user-images.githubusercontent.com/111701513/210265982-0a403cf7-ba59-43d4-a8c9-fe3e10da2f3c.png)

#### example and use case 3: multilanguage support with Markdown
![image](https://user-images.githubusercontent.com/111701513/210294387-e8c81077-7092-4d3b-88b9-01b958fcb561.png)

*An advantage that I see in your request could be the multilanguage support, instead of creating a README.md in English, we could create several readme tabs with support for several languages.*

**Reply by Thiamath** [reply]
1 and 3 sound great!

I don't understand much about the 2nd... sorry.

**Reply by ghost** [reply]
Hi Thiamath. thank you for feedback.

> I don't understand much about the 2nd... sorry.

Is for you to add categories of links, for example: social link, media link etc. So... you can... add social media links or guides with Markdown.

**Comment by mahozad** [comment]
Maybe related: [Ability to add tabbed code snippets in Markdown](https://github.com/community/community/discussions/17381).

**Comment by dwmkerr** [comment]
This would be amazing, e.g. for projects where we want to show how to setup configuration files in multiple formats (json/yaml) or show different languages (javascript/typescript).

**Comment by SKYBITDev3** [comment]
Is there any working syntax or workaround?

**Comment by numandev1** [comment]
is there any chance?

**Comment by krupis** [comment]
I am also interested in this. Is there a way to create tabs in markdown document?

**Comment by LatentArchitect** [comment]
Add me to the list of interested people

**Comment by singhk97** [comment]
Add me to list of interested people

**Comment by biscuit314** [comment]
I vote for this feature!

**Comment by jasonh-edfi** [comment]
Wow - this would really help our technical documentation for various platforms (Windows/Ubuntu/Docker) and supported database (MS SQL/Postgres/etc)!  Hoping this goes into the spec and we can take advantage of  `### tab` for tabbed content.

**Comment by atlamors** [comment]
Any update on this?

**Comment by NamNamIoT** [comment]
Any update on this?

**Comment by RockyMM** [comment]
I'm guessing this overlaps with [the Default Community health files](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/creating-a-default-community-health-file#supported-file-types).

Y'all should instead look into that option which is already supported.

**Reply by katomaso** [reply]
Interested as well!

**Comment by bjcresswell** [comment]
Another vote for this feature please!

**Comment by rockyshi1993** [comment]
This problem can be solved using [markdown-it-codetabs ](https://github.com/cncws/markdown-it-codetabs)

**Reply by rockyshi1993** [reply]
I re-added an npm package [markdow-it-code-tabs](https://www.npmjs.com/package/markdow-it-code-tabs)

**Reply by RockyMM** [reply]
How do you mean to solve displaying tabs in GitHub with an external npm package? Could explain your idea in a bit more detail?

**Comment by bkkflynnsta** [comment]
Disappointed along with all my fellow developers, but here's an ugly `hack`/ `workaround` that might bridge the gap for some until this *table stakes* item is delivered (if ever...)

You might be able to get by with the currently-available `<details> ... </details>` blocks, for example:

```
<details>
  <summary>NodeJS example</summary>
  *some NodeJS code here...*
</details>

<details>
  <summary>RoR example</summary>
  *some Ruby on Rails code here...*
</details>
```

Which would produce:

<details>
  <summary>NodeJS example</summary>
  *some NodeJS code here...*
</details>

<details>
  <summary>RoR example</summary>
  *some Ruby on Rails code here...*
</details>

**Reply by bordumb** [reply]
@bkkflynnsta I'd consider that the best workaround 

But from a UX perspective, it means all data is initially hidden from readers

Tabs are nice because it means at least something is shown up front
